modules/light-clients/08-wasm/keeper/keeper.go

```diff
@@ -0,0 +1,189 @@
+package keeper
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"strings"
+
+	wasmvm "github.com/CosmWasm/wasmvm"
+
+	errorsmod "cosmossdk.io/errors"
+	storetypes "github.com/cosmos/cosmos-sdk/store/types"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/cosmos/ibc-go/modules/light-clients/08-wasm/internal/ibcwasm"
+	"github.com/cosmos/ibc-go/modules/light-clients/08-wasm/types"
+	clienttypes "github.com/cosmos/ibc-go/v7/modules/core/02-client/types"
+)
+
+// Keeper defines the 08-wasm keeper
+type Keeper struct {
+	// implements gRPC QueryServer interface
+	types.QueryServer
+
+	cdc    codec.BinaryCodec
+	wasmVM ibcwasm.WasmEngine
```

Comment:
is this ref needed on the keeper? 

On the main line I think we generally do `ibcwasm.GetVm()`

Reply:
Ah, let's leave this for a follow up backport if that resolves it. 

Hard to gauge what exactly is expected in here off the bat